### PR TITLE
[#5081] improvement(docs): Add the document about cloud storage fileset.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add a document about S3, GCS and OSS filesets.

### Why are the changes needed?

For better user experience.

Fix: #5081

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

N/A.

### CODE REVIEW

File: docs/hadoop-catalog.md [hunk 1]
```diff
@@ -25,19 +25,30 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `oss-secret-access-key`                            | The secret key of the Aliyun oss.                                                                                                                                                                                                                                                                             | (none)          | Yes if use oss                                              | 0.7.0-incubating | 
+
+:::warning
+Since Gravitino 0.7.0-incubating, the Hadoop catalog supports S3, GCS, and OSS. To enable the support for these object storages, you need to put the corresponding bundle jar like `gravitino-aws-bundle-{version}.jar`, `gravitino-gcs-bundle-{version}.jar`, `gravitino-oss-bundle-{version}.jar`
+into the `$GRAVITINO_HOME/catalogs/hadoop/libs` directory and set the corresponding configurations.
+:::
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:50]: Can you please separate the FS support into different table with different headings, like HDFS, S3, xxx.

Comment (yuqi1129) [reply docs/hadoop-catalog.md:50]: I was hesitate to add it separately as some properties like `filesystem-providers` are shared by several cloud storage, what about adding the following content

```markdown

### Properties required to create a local file system Hadoop Catalog

No additional properties are required to create a local file system Hadoop catalog. 

### Properties required to create an HDFS Hadoop Catalog

Similarly, no additional properties are required to create an HDFS Hadoop catalog if the target HDFS cluster is not Kerberos-enabled. If the target HDFS cluster is Kerberos-enabled, the following properties are required:
- `authentication.type`: The type of authentication for the Hadoop catalog. The value should be `kerberos`.
- `authentication.kerberos.principal`: The principal of the Kerberos authentication.
- `authentication.kerberos.keytab-uri`: The URI of the keytab for the Kerberos authentication.

### Properties required to create an S3 Hadoop Catalog
To create an S3 Hadoop catalog, you need to set the following properties:
- `filesystem-providers`: The value should be `s3` or a comma-separated list of that contains `s3` like `builtin-local,s3`. 
- `s3-endpoint`: The endpoint of the AWS S3.
- `s3-access-key-id`: The access key of the AWS S3.
- `s3-secret-access-key`: The secret key of the AWS S3.

At the same time, you need to put the corresponding bundle jar `gravitino-aws-bundle-{version}.jar` into the `$GRAVITINO_HOME/catalogs/hadoop/libs` directory.

### Properties required to create a GCS Hadoop Catalog
To create a GCS Hadoop catalog, you need to set the following properties:
- `filesystem-providers`: The value should be `gcs` or a comma-separated list of that contains `gcs` like `builtin-local,gcs`.
- `gcs-service-account-file`: The path of the GCS service account JSON file.

At the same time, you need to put the corresponding bundle jar `gravitino-gcs-bundle-{version}.jar` into the `$GRAVITINO_HOME/catalogs/hadoop/libs` directory.

### Properties required to create an OSS Hadoop Catalog
To create an OSS Hadoop catalog, you need to set the following properties:
- `filesystem-providers`: The value should be `oss` or a comma-separated list of that contains `oss` like `builtin-local,oss`.
- `oss-endpoint`: The endpoint of the Aliyun OSS.
- `oss-access-key-id`: The access key of the Aliyun OSS.
- `oss-secret-access-key`: The secret key of the Aliyun OSS.

At the same time, you need to put the corresponding bundle jar `gravitino-oss-bundle-{version}.jar` into the `$GRAVITINO_HOME/catalogs/hadoop/libs` directory.
```

Comment (yuqi1129) [reply docs/hadoop-catalog.md]: Updated.

File: docs/hadoop-catalog.md [hunk 2]
```diff
@@ -25,19 +25,77 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+
+#### S3 fileset
+
+| Configuration item             | Description                                                                                                                                                                                                                  | Default value   | Required                   | Since version    |
+|--------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|----------------------------|------------------|
+| `fs.gvfs.filesystem.providers` | The file system providers. Set it to `s3` if it's a S3 fileset                                                                                                                                                               | (none)          | Yes                        | 0.7.0-incubating |
```
Comment (FANNG1) [inline docs/hadoop-catalog.md:49]: does it support multi-provider? I couldn't find it in the description.

File: docs/hadoop-catalog.md [hunk 3]
```diff
@@ -25,19 +25,77 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `default-filesystem-provider`  | The name default filesystem providers of this Hadoop catalog if users do not specify the scheme in the URI. Default value is `builtin-local`, for S3, if we set this value, we can omit the prefix 'oss://' in the location. | `builtin-local` | No                         | 0.7.0-incubating |
+| `s3.endpoint`                  | The endpoint of the AWS s3.                                                                                                                                                                                                  | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+| `s3.access.key.id`             | The access key of the AWS s3.                                                                                                                                                                                                | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
```
Comment (FANNG1) [inline docs/hadoop-catalog.md:52]: I remember fileset use a unified S3 configuration like Iceberg, something like `s3-acess-key-id`, YES?

File: docs/hadoop-catalog.md [hunk 4]
```diff
@@ -25,19 +25,77 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `s3.secret.access.key`         | The secret key of the AWS s3.                                                                                                                                                                                                | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+
+At the same time, you need to place the corresponding bundle jar `gravitno-aws-bundle-{version}.jar` in the Hadoop environment.
```
Comment (FANNG1) [inline docs/hadoop-catalog.md:55]: could you provide the bundle jar download address?

File: docs/hadoop-catalog.md [hunk 5]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `filesystem-providers`        | The file system providers to add. Set it to `gs` if it's a gcs fileset or a comma separated string that contains `gs` like `gs,s3` to support multiple kind of fileset including `gs`.                                       | (none)          | Yes                        | 0.7.0-incubating |
+| `default-filesystem-provider` | The name default filesystem providers of this Hadoop catalog if users do not specify the scheme in the URI. Default value is `builtin-local`, for GCS, if we set this value, we can omit the prefix 'gs://' in the location. | `builtin-local` | No                         | 0.7.0-incubating |
+| `gcs-service-account-file`    | The path of GCS service account JSON file.                                                                                                                                                                                   | (none)          | Yes if it's a gcs fileset. | 0.7.0-incubating |
+
+In the meantime, you need to place the corresponding bundle jar [gravitno-gcp-bundle-{version}.jar](https://repo1.maven.org/maven2/org/apache/gravitino/gcp-bundle/) in the Hadoop environment.
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:66]: You'd better specifically tell user where to put the jar, not vaguely saying "in the Hadoop environment".

File: docs/hadoop-catalog.md [hunk 6]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `oss-endpoint`                | The endpoint of the Aliyun oss.                                                                                                                                                                                               | (none)          | Yes if it's a oss fileset. | 0.7.0-incubating |
+| `oss-access-key-id`           | The access key of the Aliyun oss.                                                                                                                                                                                             | (none)          | Yes if it's a oss fileset. | 0.7.0-incubating |
+| `oss-secret-access-key`       | The secret key of the Aliyun oss.                                                                                                                                                                                             | (none)          | Yes if it's a oss fileset. | 0.7.0-incubating |
+
+In the meantime, you need to place the corresponding bundle jar [gravitno-aliyun-bundle-{version}.jar](https://repo1.maven.org/maven2/org/apache/gravitino/aliyun-bundle/) in the Hadoop environment.
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:78]: Also here.

File: docs/hadoop-catalog.md [hunk 7]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+
+
+:::note
+- Gravitino contains builtin file system providers for local file system(`builtin-local`) and HDFS(`builtin-hdfs`), that is to say if `filesystem-providers` is not set, Gravitino will still support local file system and HDFS. Apart from that, you can set the `filesystem-providerss` to support other file systems like S3, GCS, OSS or custom file system.
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:82]: You need to add a chapter to explain file system providers, I don't think users will know this without any explanation.

Comment (jerryshao) [reply docs/hadoop-catalog.md:82]: Nevermind, I see you already explain this in the below.

File: docs/how-to-use-gvfs.md [hunk 8]
```diff
@@ -49,22 +49,69 @@ the path mapping and convert automatically.
...
+| `s3-endpoint`                  | The endpoint of the AWS s3.                                                                                                                                                           | (none)        | Yes if it's a s3 fileset. | 0.7.0-incubating |
+| `s3-access-key-id`             | The access key of the AWS s3.                                                                                                                                                         | (none)        | Yes if it's a s3 fileset. | 0.7.0-incubating |
+| `s3-secret-access-key`         | The secret key of the AWS s3.                                                                                                                                                         | (none)        | Yes if it's a s3 fileset. | 0.7.0-incubating |
+
+At the same time, you need to place the corresponding bundle jar [gravitino-aws-bundle-{version}.jar](https://repo1.maven.org/maven2/org/apache/gravitino/aws-bundle/) in the Hadoop environment.
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md:81]: You'd better give an example of how to use gvfs with s3 or gcs, not vaguely saying "in the Hadoop environment".

File: docs/how-to-use-gvfs.md [hunk 9]
```diff
@@ -335,17 +382,24 @@ to recompile the native libraries like `libhdfs` and others, and completely repl
...
+| `gcs_service_account_file` | The path of GCS service account JSON file.                                                                                                               | (none)        | Yes if it's a gcs fileset         | 0.7.0-incubating |
+| `oss_endpoint`             | The endpoint of the Aliyun oss.                                                                                                                          | (none)        | Yes if it's a oss fileset         | 0.7.0-incubating |
+| `oss_access_key_id`        | The access key of the Aliyun oss.                                                                                                                        | (none)        | Yes if it's a oss fileset         | 0.7.0-incubating |
+| `oss_secret_access_key`    | The secret key of the Aliyun oss.                                                                                                                        | (none)        | Yes if it's a oss fileset         | 0.7.0-incubating |
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md:402]: You'd better also split into several tables like Java gvfs. also please give an example here.

Comment (yuqi1129) [reply docs/how-to-use-gvfs.md]: done.

File: docs/how-to-use-gvfs.md (hunk 8 above)
Comment (yuqi1129) [reply docs/how-to-use-gvfs.md]: done.

File: docs/hadoop-catalog.md (hunk 6 above)
Comment (yuqi1129) [reply docs/hadoop-catalog.md]: done.

File: docs/hadoop-catalog.md [hunk 10]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+
+#### S3 fileset
+
+| Configuration item             | Description                                                                                                                                                                           | Default value   | Required                   | Since version    |
+|--------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|----------------------------|------------------|
+| `filesystem-providers`         | The file system providers to add. Set it to `s3` if it's a s3 fileset or a comma separated string that contains `s3` like `gs,s3` to support multiple kind of fileset including `s3`. | (none)          | Yes                        | 0.7.0-incubating |
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:49]: “..., or a comma separated ... to support multiple kinds of...”

File: docs/hadoop-catalog.md [hunk 11]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `default-filesystem-provider`  | The name default filesystem providers of this Hadoop catalog if users do not specify the scheme in the URI. Default value is `builtin-local`, for S3, if we set this value, we can omit the prefix 'oss://' in the location. | `builtin-local` | No                         | 0.7.0-incubating |
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:50]: Why you mentioned S3 here, but the prefix is "oss://"?

File: docs/hadoop-catalog.md [hunk 12]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `s3-endpoint`                  | The endpoint of the AWS s3.                                                                                                                                                           | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+| `s3-access-key-id`             | The access key of the AWS s3.                                                                                                                                                         | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+| `s3-secret-access-key`         | The secret key of the AWS s3.                                                                                                                                                         | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+
+At the same time, you need to place the corresponding bundle jar [gravitno-aws-bundle-{version}.jar](https://repo1.maven.org/maven2/org/apache/gravitino/aws-bundle/) in the directory ${GRAVITINO_HOME}/catalogs/hadoop/libs.
+
+
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:57]: Remove the blank line here and below.

File: docs/hadoop-catalog.md [hunk 13]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `s3-endpoint`                  | The endpoint of the AWS s3.                                                                                                                                                           | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+| `s3-access-key-id`             | The access key of the AWS s3.                                                                                                                                                         | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+| `s3-secret-access-key`         | The secret key of the AWS s3.                                                                                                                                                         | (none)          | Yes if it's a s3 fileset.  | 0.7.0-incubating |
+
+At the same time, you need to place the corresponding bundle jar [gravitno-aws-bundle-{version}.jar](https://repo1.maven.org/maven2/org/apache/gravitino/aws-bundle/) in the directory ${GRAVITINO_HOME}/catalogs/hadoop/libs.
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:55]: "gravitino"

File: docs/hadoop-catalog.md [hunk 14]
```diff
@@ -25,19 +25,88 @@ Hadoop 3. If there's any compatibility issue, please create an [issue](https://g
...
+| `filesystem-providers`        | The file system providers to add. Set it to `gs` if it's a gcs fileset or a comma separated string that contains `gs` like `gs,s3` to support multiple kind of fileset including `gs`.                                       | (none)          | Yes                        | 0.7.0-incubating |
+| `default-filesystem-provider` | The name default filesystem providers of this Hadoop catalog if users do not specify the scheme in the URI. Default value is `builtin-local`, for GCS, if we set this value, we can omit the prefix 'gs://' in the location. | `builtin-local` | No                         | 0.7.0-incubating |
+| `gcs-service-account-file`    | The path of GCS service account JSON file.                                                                                                                                                                                   | (none)          | Yes if it's a gcs fileset. | 0.7.0-incubating |
+
+In the meantime, you need to place the corresponding bundle jar [gravitno-gcp-bundle-{version}.jar](https://repo1.maven.org/maven2/org/apache/gravitino/gcp-bundle/) in the directory ${GRAVITINO_HOME}/catalogs/hadoop/libs.
```
Comment (jerryshao) [inline docs/hadoop-catalog.md:66]: "gravitino"

File: docs/hadoop-catalog.md (hunk 10 above)
Comment (yuqi1129) [reply docs/hadoop-catalog.md]: fixed

File: docs/hadoop-catalog.md (hunk 11 above)
Comment (yuqi1129) [reply docs/hadoop-catalog.md]: fixed

**Comment by yuqi1129** [comment]
all comments have been resolved.

File: docs/how-to-use-gvfs.md [hunk 15]
```diff
@@ -49,22 +49,69 @@ the path mapping and convert automatically.
...
+| `oss-secret-access-key`         | The secret key of the Aliyun oss.                                                                                                                                                             | (none)        | Yes if it's a oss fileset. | 0.7.0-incubating |
+
+In the meantime, you need to place the corresponding bundle jar [gravitino-aliyun-bundle-{version}.jar](https://repo1.maven.org/maven2/org/apache/gravitino/aliyun-bundle/) in the Hadoop environment(typically located in `${HADOOP_HOME}/share/hadoop/common/lib/`).
+
+#### Custom fileset 
+Since 0.7.0-incubating, users can define their own fileset type and configure the corresponding properties, for more, please refer to [Custom Fileset](./hadoop-catalog.md#how-to-custom-your-own-hcfs-file-system-fileset).
+So, If you want to access the custom fileset through GVFS, you need to configure the corresponding properties.
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md:107]: "If you want to ..."

File: docs/how-to-use-gvfs.md [hunk 16]
```diff
@@ -76,9 +123,21 @@ You can configure these properties in two ways:
...
+    conf.set("s3-endpoint", "http://localhost:9000");
+    conf.set("s3-access-key-id", "minio");
+    conf.set("s3-secret-access-key", "minio123"); 
+   
     Path filesetPath = new Path("gvfs://fileset/test_catalog/test_schema/test_fileset_1");
     FileSystem fs = filesetPath.getFileSystem(conf);
     ```
+   
+:::note
+If you want to access the s3, gcs, oss or custom fileset through GVFS, apart from the above properties, you need to place the corresponding bundle jar in the Hadoop environment.
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md:138]: "S3, GCS, OSS"

File: docs/how-to-use-gvfs.md [hunk 17]
```diff
@@ -102,6 +161,24 @@ You can configure these properties in two ways:
         <name>fs.gravitino.client.metalake</name>
         <value>test_metalake</value>
       </property>
+   
+      <!-- Optional. It's only for s3 catalog, for GCs and OSS catalog, you should set the corresponding properties. -->
```
Comment (jerryshao) [inline docs/how-to-use-gvfs.md]: "GCS"

**Review by jerryshao — APPROVED** [review]

File: docs/hadoop-catalog.md [hunk 18]
```diff
@@ -44,46 +44,43 @@ Apart from the above properties, to access fileset like HDFS, S3, GCS, OSS or cu
...
 |-------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|----------------------------|------------------|
-| `filesystem-providers`        | The file system providers to add. Set it to `gs` if it's a gcs fileset or a comma separated string that contains `gs` like `gs,s3` to support multiple kind of fileset including `gs`.                                       | (none)          | Yes                        | 0.7.0-incubating |
+| `filesystem-providers`        | The file system providers to add. Set it to `gs` if it's a gcs fileset, a comma separated string that contains `gs` like `gs,s3` to support multiple kinds of fileset including `gs`.                                        | (none)          | Yes                        | 0.7.0-incubating |
```
[the remark this answers is not in the document]
Comment (jerryshao) [reply docs/hadoop-catalog.md]: Sorry seems like the trailing whitespace is mainly for the table alignment, right? If so, please keep them for the ease of review.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**